module/zfs/vdev_mirror.c

```diff
@@ -309,6 +310,11 @@ vdev_mirror_map_init(zio_t *zio)
 
 			mc->mc_vd = vdev_lookup_top(spa, DVA_GET_VDEV(&dva[c]));
 			mc->mc_offset = DVA_GET_OFFSET(&dva[c]);
+			if (mc->mc_vd == NULL) {
+				kmem_free(mm, vdev_mirror_map_size(
+				    mm->mm_children));
+				return (NULL);
+			}
```

Comment:
Did you mean to remove this block now that you've updated `!spa_trust_config` -> `!spa_writeable` above?

Reply:
It's my understanding that this was left intentionally to gracefully handle the case where either the DVA references a vdev which is for some reason missing from the config or otherwise damaged.  While this shouldn't be possible, handling it only makes thinging more robust so I'd like to include it.

Reply:
@behlendorf Yes, that's what I was thinking.  Thanks.

Reply:
I believe in that case we should also set those 2 parameters like we do above:
```
zio->io_vsd = NULL;
zio->io_error = ENXIO;
```

Reply:
Yes, good catch.

Reply:
With those changes in, this LGTM.

Reply:
Thanks; I added it.